clouddq/classes/dq_target_table_utils.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from datetime import date
+
+from google.cloud import bigquery
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_target_table_from_bigquery(
+    bigquery_client: bigquery.client.Client,
+    invocation_id: str,
+    partition_date: date,
+    target_bigquery_summary_table: str,
+    dq_summary_table_name: str,
+):
+
+    default_job_config = bigquery.QueryJobConfig(use_query_cache=True)
+
+    if bigquery_client.is_table_exists(target_bigquery_summary_table):
+
+        job_config = bigquery.QueryJobConfig(
+            destination=target_bigquery_summary_table,
+            write_disposition="WRITE_APPEND",
+        )
+
+        query_string = f"""SELECT * FROM `{dq_summary_table_name}` 
+         WHERE invocation_id='{invocation_id}' 
+         and DATE(execution_ts)='{partition_date}'"""
+
+        # Start the query, passing in the extra configuration.
+        # Make an API request and wait for the job to complete
+        bigquery_client.execute_query(query_string=query_string, job_config=job_config)
```

Comment:
is this a synchronous call i.e. the execute_query blocks until the query completes? or is it asynchronous and returns a future?

Reply:
yes, I made it synchronous by chaining a .result() to client.query() so that it will wait till the BQ job is complete.